pkg/adapter/datasource.go

```diff
@@ -105,7 +106,15 @@ func (d *Datasource) GetPage(ctx context.Context, request *Request) (*Response,
 	// SCAFFOLDING #17 - pkg/adapter/datasource.go: Add any headers required to communicate with the SoR APIs.
 	// Add headers to the request, if any.
 	// req.Header.Add("Accept", "application/json")
-	// req.Header.Add("Authorization", "Bearer Token")
+
+	if request.Token == "" {
+		// Basic Authentication
+		auth := request.Username + ":" + request.Password
+		req.Header.Add("Authorization", "Basic "+base64.StdEncoding.EncodeToString([]byte(auth)))
```

Comment:
nit: I understand that we might not want to setup CI with linters on a public repo. Let's make sure to run them locally if possible before merging. Not a blocker right now. We (@r-rakshith / @richardxgao / @nholbrook) will follow up in a different PR.

```suggestion
		req.Header.Add("Authorization", "Basic " + base64.StdEncoding.EncodeToString([]byte(auth)))
```